apps/schematic/api/schematic_api/controllers/utils.py

```diff
@@ -184,3 +193,110 @@ def download_schema_file_as_jsonld(schema_url: str) -> str:
         raise InvalidSchemaURL(
             "The provided URL could not be found", schema_url
         ) from error
+
+
+def purge_synapse_cache(
+    store: SynapseStorage,
+    maximum_storage_allowed_cache_gb: float = 1,
+    minute_buffer: int = 15,
+) -> None:
+    """
+    Purge synapse cache if it exceeds a certain size. Default to 1GB.
+    Args:
+        maximum_storage_allowed_cache_gb (float): the maximum storage allowed
+          before purging cache. Default is 1 GB.
+        minute_buffer (int): All files created this amount of time or older will be deleted
+    """
+    # try clearing the cache
+    # scan a directory and check size of files
+    if os.path.exists(store.root_synapse_cache):
+        maximum_storage_allowed_cache_bytes = maximum_storage_allowed_cache_gb * (
+            1024**3
+        )
+        dir_size_bytes = check_synapse_cache_size(directory=store.root_synapse_cache)
+        # if 1 GB has already been taken, purge cache before 15 min
```

Comment:
could you also edit this comment? since minute_buffer is a parameter, and don't always clear cache before 15min

Reply:
Yep!